src/main/java/com/thebois/models/beings/AbstractBeing.java

```diff
@@ -1,24 +1,22 @@
 package com.thebois.models.beings;
 
-import java.util.Random;
+import java.util.Objects;
 
-import com.thebois.models.IDeepClonable;
 import com.thebois.models.Position;
+import com.thebois.models.beings.roles.AbstractRole;
 
 /**
  * An abstract implementation of IBeing.
  */
-public abstract class AbstractBeing implements IBeing, IDeepClonable<IBeing> {
```

Comment:
We don't want `IDeepClonable<IBeing>` because `AbstractBeing` is a base class for an Entity, and they shouldn't be immutable and therefore there is no need for it being clonable.